### ISSUE TITLE
`vue/prefer-template` tries to apply to `style scoped`

### ISSUE THREAD

**Issue by sky0matic** [vuejs/eslint-plugin-vue#2712]
**What rule do you want to change?**
`vue/prefer-template`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
The rule doesn't check for `style scoped` or there's a way to disable it on a per-line basis.

**Please provide some example code that this change will affect:**

```vue
<script setup lang="ts">
  const someWidth = computed(() => 
    // Some computation process
  );
</script>

<style scoped>
.some-element {
  width: v-bind(someWidth + 'px');
}

.some-other-element {
  /* same issue if we try to disable the rule, it doesn't seem to work */
  /* eslint-disable-next-line vue/prefer-template */
  width: v-bind(someWidth + 'px');
}
</style>
```

**What does the rule currently do for this code?**
It want us to replace `someWidth + 'px'` with : 
```
`{someWidth}px`
```
and this is not valid CSS handled by the IDE (VSCode in this case) so it shows another error.

**What will the rule do after it's changed?**
Either :
- Not react to `style scoped`
- Allow to disable the rule on the style somehow or react to the `eslint-disable-next-line` properly.

**Additional context**
The only way we are able to skip this issue entirely for now is to create a new computed which does : 
```vue
const someWidthStr = computed(() => `${someWidth.value}px`);
```
which makes us need to maintain yet another `computed` which we would prefer to avoid.

**Comment by FloEdelmann** [comment]
Oh that's a bug; the rule should only search those patterns in the `<template>` block.

**Comment by FloEdelmann** [comment]
I tried reproducing the error in the [playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNz8EOgjAMBuBXaXoBokjCxWQCR1/BCxdkVZeMbWEFNYR3l6FGb23//F/SCVsrCQUWTJ3TDVNVG4BCqrE6WltkYahNkf3Fy+rbXjkGTzw40I25ljWyr3Ett9b4JbMdnZTkG5TLpXMDk4zjBMoK8n1yCOZb+Yj81AS+tY7kctmFekqaOjIMU2DvARMwpmdlZPzjNxC5RxTEeUUDVOEW+0GTRzHhOFDmerpQn36/QJHP8wsYlFdo) but it always reports `Parsing error: t is undefined`. @ota-meshi do you know why?

**Comment by ota-meshi** [comment]
Thank you for letting me know, I fixed the playground 😉

**Comment by waynzh** [comment]
@sky0matic This seems more like a usage issue, rather than a auto-fix bug in the rule. 
According to the [Vue docs](https://vuejs.org/api/sfc-css-features.html#v-bind-in-css), "v-bind in CSS must be wrapped in quotes." 
This can be addressed by changing `width: v-bind(someWidth + 'px');` to `width: v-bind("someWidth + 'px'");`. After the auto-fix, it will be converted to 「v-bind("\`${someWidth}px\`")」, which will then function correctly without any lint errors.

**Comment by waynzh** [comment]
> this is not valid CSS handled by the IDE (VSCode in this case) so it shows another error.

So, this is why it triggers a CSS lint error, and there's also a warning indicating that the variable is declared but its value is never read.
<img width="546" alt="Image" src="https://github.com/user-attachments/assets/1970764a-1ac7-4fbf-8199-a10db2fe78c9" />

I'm going to close this issue for now, but feel free to continue the discussion if you have further questions.
